subsys/bluetooth/host/hci_core.c

```diff
@@ -1180,6 +1232,10 @@ static void enh_conn_complete(struct bt_hci_evt_le_enh_conn_complete *evt)
 		 */
 		conn = find_pending_connect(NULL);
 		if (!conn) {
+			if (IS_ENABLED(CONFIG_BT_WHITELIST)) {
+				notify_connected(NULL);
```

Comment:
@jhedberg I'm a bit unsure about this one.

Maybe it would be better to allocate a valid conn object when starting whitelist conn create procedure, so that when we get connected we won't have to disconnect because we don't have enough conn objects.

Reply:
@joerchan we should pre-allocate a bt_conn any time that we start an action that may result in a connection. This includes any time we start connectable advertising, whitelist conn create procedure, directed advertising, etc. I had this on my mind already a long time ago, but I'm not sure I ever got around to opening an issue for it.

Reply:
@jhedberg Alright. So we need connectable advertiser, and whitelist create connection. Directed advertising has it already. Possibly the old bt_le_set_auto_conn API.

Unrelated: I'm wondering if we maybe should deprecate bt_le_set_auto_conn  in favour of using whitelist.